server/src/utils.ts

```diff
@@ -99,6 +99,45 @@ export let formatUsingValidBscExePath = (
   }
 };
 
+export let createInterfaceFileUsingValidBscExePath = (
+  filePath: string,
+  cmiPath: string,
+  bscExePath: p.DocumentUri
+): execResult => {
+  let mliTempFile = createFileInTempDir(c.mliExt);
+
+  try {
+    // First, create a temporary .mli file
```

Comment:
You can just create a `.res` file and this command will generate the `.resi` file

Reply:
Do you mean to hand the `.res` to `bsc` instead of the `.cmi`?
Because that only worked for me with files which open no other modules. The `.cmi` approach always worked for me, but it yields ml syntax.

Reply:
Oh, right...

Reply:
Done now: https://github.com/rescript-lang/rescript-compiler/issues/5023#issuecomment-803730108

Reply:
Yeah, that's a lot better: b59e7d030e5349ed3a7d4e8c5a651ac81d36a0f0